### PR TITLE
feat(rust,python): Implement max/min methods for dtypes

### LINKED ISSUE #19412: `max` and `min` method of numeric dtypes
### Description

It would be great to have something like `i64::MAX` of Rust to quickly call the minimum and maximum literals for each numeric dtype.

For example in Python:

```python
pl.Int8.max # returns `pl.lit(127, pl.Int8)`
```

### PR DESCRIPTION
**Pull request by eitsupi**
Close #19412

### CODE REVIEW

File: crates/polars-core/src/datatypes/dtype.rs [hunk 1]
```diff
@@ -572,6 +572,52 @@ impl DataType {
...
+            Int64 => Scalar::from(i64::MAX),
+            #[cfg(feature = "dtype-u8")]
+            UInt8 => Scalar::from(u8::MAX),
+            #[cfg(feature = "dtype-u16")]
+            UInt16 => Scalar::from(u16::MAX),
+            UInt32 => Scalar::from(u32::MAX),
+            UInt64 => Scalar::from(u64::MAX),
+            Float32 => Scalar::from(f32::INFINITY),
+            Float64 => Scalar::from(f64::INFINITY),
+            dt => polars_bail!(ComputeError: "cannot determine upper bound for dtype `{}`", dt),
```
Comment (mcrumiller) [inline crates/polars-core/src/datatypes/dtype.rs:593]: I think here we can use something like [`NaiveDate::MAX()`](https://docs.rs/chrono/latest/chrono/naive/struct.NaiveDate.html#associatedconstant.MAX) and `MIN` from chrono here to get date limits, and likewise for `NaiveDateTime`. As an example, the naive date max value is `95026236` and we can see we hit that limit here as well:

```python
import polars as pl

pl.Series([95026236]).cast(pl.Date)  # works
pl.Series([95026237]).cast(pl.Date)  # PanicException: out-of-range date
```

Comment (eitsupi) [reply crates/polars-core/src/datatypes/dtype.rs:593]: Good point. Perhaps could be discussed in a follow-up issue?

I don't know if we should include chrono limitations here or not.

**Review by coastalwhite — APPROVED** [review]

**Comment by alexander-beedie** [comment]
Nice one; we should probably think about deprecating `upper_bound` and `lower_bound` on `Series` once this merges and simply rely on the `DataType` min/max. Seems a more natural place to get this info 🤔

**Comment by eitsupi** [comment]
This can be merged?

**Pull request merged**